envoy/listener.go

```diff
@@ -0,0 +1,38 @@
+package envoy
+
+import (
+	"github.com/Nitro/sidecar/catalog"
+)
+
+// Listener is an internal Sidecar listener that will be hooked up when
+// config.Envoy.UseGRPCAPI is true. It only needs to know when the state
```

Comment:
Hmm, is this thing keeping a separate state cache? Can't we just serve from our own catalog on the fly like we do now? I'm nervous about two copies of the state for several reasons.

Reply:
Ah yes, I re-read how that thing works. OK. Well so it goes.

Reply:
I could endeavour to provide an alternative implementation to [`SnapshotCache`](https://github.com/envoyproxy/go-control-plane/blob/4474ca038d9067f2af2c812649a30252c7c54c47/pkg/cache/simple.go#L38) which performs polling, but that would be less efficient than the current implementation. I believe our [`ServiceChanged`](https://github.com/Nitro/sidecar/blob/9e99a8496e065f2099a58618ebabb912a07ae02c/catalog/services_state.go#L191) event is reliable and, since we don't care about the contents of the event, we shouldn't bump into #50, but let's talk about it :)

Reply:
Now that I looked into it, this commit from #50 might be necessary: https://github.com/Nitro/sidecar/pull/50/commits/ece1e69f36605a76d3177644b9bee2d27597987d

Reply:
Seems fine as-is